src/SqlAsyncEnumerable.cs

```diff
@@ -72,13 +74,15 @@ public SqlAsyncEnumerator(SqlConnection connection, SqlAttribute attribute)
 
             /// <summary>
             /// Closes the SQL connection and resources associated with reading the results of the query
+            /// Set _isDisposed to true so any further MoveNextAsync calls will not attempt to read from a disposed reader. 
```

Comment:
I'd put a comment down in GetNextRowAsync instead and mention that you were seeing it called after DisposeAsync was called, it'd be easy to miss this comment currently.